internal/utils.go

```diff
@@ -42,14 +45,49 @@ func filterByRegex(pattern string, fi []*funcInfo) ([]*funcInfo, error) {
 
 func getFilename(filePath string) (string, error) {
 	dir, file := filepath.Split(filePath)
-	pkg, err := build.Import(dir, ".", build.FindOnly)
-	if err != nil {
-		return "", err
-	}
+	// TODO: refactor this to simpler approach
+	var pkg *build.Package
+	var err error
+	fext := strings.Split(file, ".")[1]
+	if strings.EqualFold(fext, "go") {
+		pkg, err = build.Import(strings.TrimSuffix(dir, "/"), ".", build.FindOnly)
+		if err != nil {
+			if strings.Contains(err.Error(), "go get") {
+				if err := importModule(strings.TrimSuffix(dir, "/")); err != nil {
+					return "", err
+				}
+			}
+			return "", err
+		}
+
+	} else {
+		pkg, err = build.ImportDir(dir, build.FindOnly)
+		if err != nil {
+			_, err = build.Import(dir, ".", build.FindOnly)
+			if err != nil {
+				if strings.Contains(err.Error(), "add") {
+					if err := importModule(strings.TrimSuffix(dir, "/")); err != nil {
+						return "", err
+					}
+					return "", err
+				}
 
+			}
+			return "", err
+		}
+	}
 	return filepath.Join(pkg.Dir, file), nil
 }
 
+func importModule(s string) error {
+	cmd := exec.Command("go")
+	cmd.Args = append(cmd.Args, "get", fmt.Sprintf("%s@latest", s))
```

Comment:
I'm not sure on pinning dependencies to `@latest`. Is there any approach that we could take?

Reply:
not sure what is your thoughts on pinning version other than latest.

Reply:
Users can prefer to pin to a certain version for some dependencies; fetching the `latest` version can lead to the modification of `go.mod`; which is not desired in many situations.

The tool should not change the dependencies on its own - we should let the users decide what's best in their use case.

Reply:
@burntcarrot sorry for the hibernation with PR , did get some time and worked it out, https://github.com/burntcarrot/samosa/pull/4/commits/3fec24c5b023c6275171e28d25f115d1fcd5b54b this is the latest version with all changes
refactoring.